File: client/projects/Forge/src/main/java/falcun/net/api/gui/effects/PinBottom.java
package falcun.net.api.gui.effects;
import falcun.net.api.gui.components.*;

public class PinBottom extends Effect {

	Component pinner;

	public PinBottom(Component pinner) {
		this.pinner = pinner;
	}

	@Override
	public void draw(int mX, int mY, Component component, Phase phase) {
		if(phase == Phase.AFTER) {
			return;
		}
		component.region.y = pinner.region.getBottom();
	}
}

File: game/src/main/scala/falcun/net/api/gui/effects/PinBottom.scala
package falcun.net.api.gui.effects

import com.mojang.realmsclient.gui.ChatFormatting
import falcun.net.api.gui.components._
import falcun.net.api.gui.effects.Effect.Phase

import java.util.function.BiConsumer
import java.util.function.Supplier
import java.util.function.Consumer
import falcun.net.api.gui.region.GuiRegion
import net.minecraft.client.Minecraft
import org.lwjgl.input.Mouse


class PinBottom(var pinner: Component) extends Effect {
  override def draw(mX: Int, mY: Int, component: Component, phase: Effect.Phase): Unit = {
    if (phase eq Phase.AFTER) return
    component.region.y = pinner.region.getBottom
  }
}
